cwms-data-api/src/main/java/cwms/cda/api/watersupply/WaterPumpDeleteController.java

```diff
@@ -0,0 +1,147 @@
+import io.javalin.plugin.openapi.annotations.OpenApiParam;
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.servlet.http.HttpServletResponse;
+import org.jetbrains.annotations.NotNull;
+import org.jooq.DSLContext;
+
+
+public final class WaterPumpDeleteController extends WaterSupplyControllerBase implements Handler {
+    private static final Logger LOGGER = Logger.getLogger(WaterPumpDeleteController.class.getName());
+    private static final String PUMP_TYPE = "pump-type";
+
+    public WaterPumpDeleteController(MetricRegistry metrics) {
+        waterMetrics(metrics);
+    }
+
+    @OpenApi(
+        queryParams = {
+            @OpenApiParam(name = PUMP_TYPE, required = true, type = PumpType.class,
+                    description = "The type of pump to be removed from the contract."
+                            + " Expected values: IN, OUT, OUT BELOW"),
+            @OpenApiParam(name = DELETE, type = boolean.class, required = true,
+                    description = "Whether to delete the associated accounting data.")
+        },
+        pathParams = {
+            @OpenApiParam(name = NAME, description = "The name of the pump to be "
+                    + "removed from the specified contract.", required = true),
+            @OpenApiParam(name = OFFICE, description = "The office the project is associated with.", required = true),
+            @OpenApiParam(name = PROJECT_ID, description = "The name of the project.", required = true),
+            @OpenApiParam(name = CONTRACT_NAME, description = "The name of the contract the pump is associated with.",
+                    required = true),
+            @OpenApiParam(name = WATER_USER, description = "The name of the water user the contract "
+                    + "is associated with.", required = true)
+        },
+        responses = {
+            @OpenApiResponse(status = "404", description = "The provided combination of parameters"
+                    + " did not find any contracts."),
+            @OpenApiResponse(status = "501", description = "Requested format is not implemented.")
+        },
+        description = "Delete a pump from a contract",
+        path = "/projects/{office}/{project-id}/water-user/{water-user}/contracts/{contract-id}/pumps/{name}",
+        method = HttpMethod.DELETE,
+        tags = {TAG}
+    )
+
+    @Override
+    public void handle(@NotNull Context ctx) {
+        try (Timer.Context ignored = markAndTime(DELETE)) {
+            DSLContext dsl = getDslContext(ctx);
+            boolean deleteAccounting = Boolean.parseBoolean(ctx.queryParam(DELETE));
+            String officeId = ctx.pathParam(OFFICE);
+            String projectName = ctx.pathParam(PROJECT_ID);
+            String entityName = ctx.pathParam(WATER_USER);
+            PumpType pumpType = PumpType.valueOf(ctx.queryParam(PUMP_TYPE));
```

Comment:
hec-cwms-data-access' usage of this API already knows the name of the pump, you can save a database call by having the client pass in the pump name rather than the pump type which

Reply:
PL call in DAO to DB requires pump type as input

Reply:
yeah, the client should specify the pump type and the pump name. No need to go look up the name within CDA given the client usage.